src/aspeak/api/api.py

```diff
@@ -42,16 +45,37 @@ def __init__(self, locale: Optional[str] = None, voice: Optional[str] = None,
             self._config.set_speech_synthesis_output_format(parse_format(audio_format))
         self._synthesizer = speechsdk.SpeechSynthesizer(self._config, self._output)
 
+    def config(self):
+        html = get(GET_TOKEN,verify=False)
+        html.raise_for_status()
+        html = html.text
+        token = search(r'token: "(.+)"',html)
+        region = search(r'region: "(.+)"',html)
+        assert token is not None
+        assert region is not None
+        self.time = time()
+        print(f"region={region.group(1)} auth_token={'bearer '+token.group(1)}")
+        self._config = speechsdk.SpeechConfig(region=region.group(1),auth_token="bearer "+token.group(1))
+
+    def _chk(self):
+        now = time()
+        if now-self.time>290:
+            self.config()
```

Comment:
Simply calling `self.config` isn't enough here. `__init__` changes some properties of `self._config`.

And you need to recreate the `SpeechSynthesizer` when you change `self._config`.

Your code works for CLI but it doesn't work if aspeak is used as a python library.

Reply:
I'm trying to fix, and I tried this method through change https://github.com/flt6/tts. However, I can't access any program (Timeout...). This may be due to my own environment.